### PR TITLE
SW-4259 'Observation not scheduled' notification emails to Terraformation Contact/Support

### PR DESCRIPTION
**Pull request by karthikbtf**
- Send email notifications to Terraformation Contact (or Terraformation Support) when observations haven't been scheduled in a while
- First notification is sent when 8 weeks have elapsed since the last completed observation (or 6 weeks since the first planting)
- Second notification is sent when 16 weeks have elapsed since the last completed observation (or 14 weeks since the first planting)
- Added a support config in TerrawareServerConfig to configure a support email
- Email is sent to the org's Terraformation Contact user if one exists, otherwise to the support email (if configured)
- Added tests
- Tested locally with the test clock, had to increase test clock html input to 3 (to advance clock to 16 weeks)
- There is no app notification for this use case

### CODE REVIEW

File: src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt]: This comment seems inaccurate

File: src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/email/EmailNotificationService.kt]: I think `if (user != null)` would probably be a bit easier to read here than the `let` expression.

File: src/main/kotlin/com/terraformation/backend/email/EmailService.kt
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/email/EmailService.kt]: The use of `it` is slightly unclear here since it's in a nested lambda. Maybe introduce a name for it like `email`?

File: src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt
Comment (sgrimm) [inline src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt]: There are now enough of these that I wonder if it makes sense to refactor them to call a helper method that takes the timestamp field to update as an argument.

File: src/main/resources/i18n/Messages_en.properties
Comment (sgrimm) [inline src/main/resources/i18n/Messages_en.properties]: Blank line will get removed by Phrase

**Review by sgrimm — CHANGES_REQUESTED** [review]
Just some small stuff; generally this looks fine, though I worry that we're now accumulating a lot of nearly-identical notification functions.

**Comment by karthikbtf** [comment]
> Just some small stuff; generally this looks fine, though I worry that we're now accumulating a lot of nearly-identical notification functions.

Thanks again for the feedback, applied the changes. Also modified the parent PR to remove the various almost identical functions, which I agree was not too pretty. Updated this PR to leverage the criteria model in the parent PR.

**Review by sgrimm — APPROVED** [review]
Looks good, but of course if you go with my naming suggestions on the parent PR, this one should change too. If not, this is good to go as-is once the parent is merged.

**Pull request merged**
